include/picongpu/particles/debyeLength/Check.hpp

```diff
@@ -0,0 +1,180 @@
+                    pmacc::mpi::MPIReduce reduce;
+                    reduce(
+                        pmacc::nvidia::functors::Add(),
+                        &globalResult.numViolatingSupercells,
+                        &localResult.numViolatingSupercells,
+                        1,
+                        pmacc::mpi::reduceMethods::AllReduce());
+                    reduce(
+                        pmacc::nvidia::functors::Add(),
+                        &globalResult.numActiveSupercells,
+                        &localResult.numActiveSupercells,
+                        1,
+                        pmacc::mpi::reduceMethods::AllReduce());
+                    if(globalResult.numActiveSupercells)
+                    {
+                        auto const ratioFailing = static_cast<float_64>(globalResult.numViolatingSupercells)
+                            / static_cast<float_64>(globalResult.numActiveSupercells);
+                        /* Due to random nature of temperature and estimates used internally,
+                         * we allow a relatively small number of supercells to fail the check
+                         */
+                        auto const thresholdFailing = 0.1;
+                        if(ratioFailing <= thresholdFailing)
+                            return Status::Passed;
+                        else
+                            return Status::Failed;
+                    }
+                    else
+                        return Status::Skipped;
+                }
+
+            } // namespace detail
+
+            /** Check Debye length resolution
+             *
+             * The check is currently done only for species called "e" and is supposed
+             * to be called just after the particles are initialized as start of a
+             * simulation. The results are output to log< picLog::PHYSICS >.
+             *
+             * This function must be called from all MPI ranks.
+             *
+             * @param cellDescription mapping for kernels
+             */
+            HINLINE void check(MappingDesc const cellDescription)
+            {
+                bool isPrinting = (Environment<simDim>::get().GridController().getGlobalRank() == 0);
+                using ElectronSpecies = pmacc::particles::meta::FindByNameOrType_t<
+                    VectorAllSpecies,
+                    PMACC_CSTRING("e"),
+                    pmacc::errorHandlerPolicies::ReturnType<void>>;
+                bool isElectronsFound = !std::is_same<ElectronSpecies, void>::value;
+                if(isElectronsFound)
+                {
+                    auto status = detail::checkDebyeLength<ElectronSpecies>(cellDescription);
+                    if(isPrinting)
+                        switch(status)
+                        {
+                        case detail::Status::Passed:
+                            log<picLog::PHYSICS>("Estimated Debye length for species \"e\" is resolved by the grid");
```

Comment:
Could we please always output the calculated values and the required? IN case the user is using a too high temperature he is able to see it by this output.

Reply:
I do not see how to output it clearly. Please see my answer to your previous question.